tests/test_tasks.py

```diff
@@ -1,104 +1,242 @@
-import lm_eval.tasks as tasks
-
 import pytest
 from itertools import islice
+import lm_eval.tasks as tasks
+from tests.extra.test_utils import load_changed_files, parser
+from typing import List, ClassVar
+import os
 
 
-@pytest.mark.parametrize("taskname,task_class", tasks.TASK_REGISTRY.items())
-def test_basic_interface(taskname, task_class):
-    print("Evaluating task", taskname)
-    task = task_class()
+@pytest.fixture()
+def any_new_tasks(request) -> bool:
+    return request.config.getoption("--new_task")
 
-    assert task.has_training_docs() in [True, False]
-    assert task.has_validation_docs() in [True, False]
-    assert task.has_test_docs() in [True, False]
 
-    assert isinstance(task.aggregation(), dict)
-    assert isinstance(task.higher_is_better(), dict)
-    assert task.aggregation().keys() == task.higher_is_better().keys()
+# ["arc_easy] else get list of new tasks
+def new_tasks(any_new_tasks: bool) -> List[str]:
+    FILENAME = ".github/outputs/tasks_all_changed_and_modified_files.txt"
+    if any_new_tasks and os.path.exists(FILENAME):
+        return [parser(load_changed_files(FILENAME))]
+    elif os.getenv("API") is not None:
+        return ["arc_easy", "hellaswag", "piqa", "wikitext"]
```

Comment:
change these tasks if you want. case where api/*.py are changed.

Reply:
LGTM!